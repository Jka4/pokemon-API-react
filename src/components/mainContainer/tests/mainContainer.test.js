import React from 'react';
import MainContainer from '@MainContainer';
import renderer, { act } from 'react-test-renderer';
import { BrowserRouter as Router } from "react-router-dom";

it('MainContainer renders correctly', async () => {

	act(() => {
		const tree = renderer
			.create(<Router><MainContainer /></Router >)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});



});
