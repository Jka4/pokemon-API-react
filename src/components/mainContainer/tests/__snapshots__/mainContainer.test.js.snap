// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer renders correctly 1`] = `
<section
  className="main"
>
  <div
    className="topLine"
  >
    <div
      className="player"
    >
      <button
        className="playButton"
        onClick={[Function]}
      >
        PLAY
      </button>
    </div>
    <div
      className="search"
    >
      <input
        aria-label="Search"
        autoComplete="off"
        id="searchInput"
        name="searchInput"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Pickachu, 123, 900g"
        type="search"
      />
      <div
        className="searchResults"
      >
        <ul
          className="searchList"
        />
      </div>
    </div>
  </div>
  <div
    className="getButtonLine"
  >
    <div
      className="Tilt"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      style={Object {}}
    >
      <div
        className="Tilt-inner"
      >
        <div
          aria-busy="true"
          className=""
          style={Object {}}
        >
          <svg
            aria-label="audio-loading"
            height={120}
            viewBox="0 0 38 38"
            width={150}
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs>
              <linearGradient
                id="a"
                x1="8.042%"
                x2="65.682%"
                y1="0%"
                y2="23.865%"
              >
                <stop
                  offset="0%"
                  stopColor="red"
                  stopOpacity="0"
                />
                <stop
                  offset="63.146%"
                  stopColor="red"
                  stopOpacity=".631"
                />
                <stop
                  offset="100%"
                  stopColor="red"
                />
              </linearGradient>
            </defs>
            <g
              fill="none"
              fillRule="evenodd"
            >
              <g
                transform="translate(1 1)"
              >
                <path
                  d="M36 18c0-9.94-8.06-18-18-18"
                  id="Oval-2"
                  stroke="red"
                  strokeWidth="2"
                >
                  <animateTransform
                    attributeName="transform"
                    dur="0.9s"
                    from="0 18 18"
                    repeatCount="indefinite"
                    to="360 18 18"
                    type="rotate"
                  />
                </path>
                <circle
                  cx="36"
                  cy="18"
                  fill="#fff"
                  r={1}
                >
                  <animateTransform
                    attributeName="transform"
                    dur="0.9s"
                    from="0 18 18"
                    repeatCount="indefinite"
                    to="360 18 18"
                    type="rotate"
                  />
                </circle>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Button"
  >
    <button
      id="getPokemon"
      onClick={[Function]}
    >
      GET A RANDOM POKEMON
    </button>
    <button
      id="clear"
      onClick={[Function]}
    >
      CLEAR
    </button>
    <a
      aria-current={null}
      href="/allPokemons"
      onClick={[Function]}
    >
      <button
        id="getAllPokemon"
      >
        GET ALL POKEMON
      </button>
    </a>
  </div>
  <div
    className="itemList"
  >
    <div
      className="cardsContainer"
    />
  </div>
</section>
`;
